src/core/thread.d

```diff
@@ -4427,13 +4430,26 @@ private:
             {
                 m_ctxt.bstack = m_pmem + sz;
                 m_ctxt.tstack = m_pmem + sz;
+                void* guard = m_pmem;
             }
             else
             {
                 m_ctxt.bstack = m_pmem;
                 m_ctxt.tstack = m_pmem;
+                void* guard = m_pmem + sz - PAGESIZE;
             }
             m_size = sz;
+
+            static if( __traits( compiles, mmap ) )
+            {
+                // Mark end of stack
+                for ( ubyte* g = cast(ubyte*)guard; g < guard + PAGESIZE; g+= 32)
+                    g[0 .. 32] = cast(ubyte[]) "END OF FIBER -- END OF FIBER -- ";
+
+                // protect end of stack
+                if ( mprotect(guard, PAGESIZE, PROT_NONE) == -1 )
+                    onOutOfMemoryError();
```

Comment:
Well, the other ones look like they allocate sth., b/c VirtualAlloc is used to change flags, and should be changed to fail instead of using onOutOfMemoryError.